tls/s2n_x509_validator.c

```diff
@@ -642,14 +629,31 @@ static S2N_RESULT s2n_x509_validator_verify_cert_chain(struct s2n_x509_validator
         /* Enable CRL validation for all certificates, not just the leaf */
         RESULT_GUARD_OSSL(X509_VERIFY_PARAM_set_flags(param, X509_V_FLAG_CRL_CHECK_ALL),
                 S2N_ERR_INTERNAL_LIBCRYPTO_ERROR);
+
+        validator->verify_cb_config.crl_callback = true;
     }
 
     /* Disabling time validation may set a NO_CHECK_TIME flag on the X509_STORE_CTX. Calling
      * X509_STORE_CTX_set_time will override this flag. To prevent this, X509_STORE_CTX_set_time is
      * only called if time validation is enabled.
      */
     if (conn->config->disable_x509_time_validation) {
-        RESULT_GUARD(s2n_x509_validator_disable_time_validation(conn, validator));
+        /* Setting an X509_STORE verify callback is not recommended with AWS-LC:
+         * https://github.com/aws/aws-lc/blob/aa90e509f2e940916fbe9fdd469a4c90c51824f6/include/openssl/x509.h#L2980-L2990
+         *
+         * If the libcrypto supports the ability to disable time validation with an X509_VERIFY_PARAM
+         * NO_CHECK_TIME flag, this method is preferred.
+         *
+         * However, older versions of AWS-LC and OpenSSL 1.0.2 do not support this flag. In this case,
+         * an X509_STORE verify callback is used. This is acceptable in older versions of AWS-LC
+         * because the versions are fixed, and updates to AWS-LC will not break the callback
+         * implementation.
+         */
+        if (s2n_libcrypto_supports_flag_no_check_time()) {
+            RESULT_GUARD(s2n_x509_validator_set_no_check_time_flag(validator));
+        } else {
+            validator->verify_cb_config.disable_time_validation = true;
```

Comment:
Rather than defining separate functions, I think I'd actually prefer the duplication? I went ahead and made that change, so let me know how that looks to you!

Reply:
I think I do like it better this way, but if you or Lindsay prefer it the other way that's fine with me as well.